vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -61,6 +70,17 @@ def auth_user_pwd(type = nil)
     [cred.userid, cred.password]
   end
 
+  def auth_user_keypair(type = nil)
+    cred =  if self.kind_of?(HostOpenstackInfra)
+              # HostOpenstackInfra is using auth key set on ext_management_system level, not individual hosts
+              self.try(:ext_management_system).try(:authentication_best_fit, type)
```

Comment:
@Fryguy actually this mixin is used both in Host and ExtManagementSystem

I think we might refactor this a bit, involving what you want to do with Provider<->Ems. Some kind of general inheritance of auth. So You can define auth on Host, EMS or Provider level and you could always overwrite it with lower layer. 

That could solve your issue, when there might be the same credentials for all EMS under provider. 

And my issue when I want the same keypair for all hosts under the EMS.

What do you think?